transport/src/test/java/io/netty/channel/pool/FixedChannelPoolTest.java

```diff
@@ -0,0 +1,207 @@
+
+        cb.group(group)
+          .channel(LocalChannel.class);
+
+        sb.group(group)
+          .channel(LocalServerChannel.class)
+          .childHandler(new ChannelInitializer<LocalChannel>() {
+              @Override
+              public void initChannel(LocalChannel ch) throws Exception {
+                  ch.pipeline().addLast(new ChannelInboundHandlerAdapter());
+              }
+          });
+
+        // Start server
+        Channel sc = sb.bind(addr).sync().channel();
+        ChannelPoolHandler<Channel, ChannelPoolKey> handler = new TestChannelPoolHandler();
+        ChannelPool<Channel, ChannelPoolKey> pool  = new FixedChannelPool<Channel, ChannelPoolKey>(
+                cb, handler, ActiveChannelHealthChecker.instance(),
+                ChannelPoolSegmentFactories.newLifoFactory(), AcquireTimeoutAction.Fail, 500,
+                1, Integer.MAX_VALUE);
+
+        ChannelPoolKey key = new DefaultChannelPoolKey(addr);
+        Channel channel = pool.acquire(key).sync().getNow();
+        Future<Channel> future = pool.acquire(key);
+        try {
+            future.sync();
+        } finally {
+            sc.close().sync();
+            channel.close().sync();
+            group.shutdownGracefully();
+        }
+    }
+
+    @Test
+    public void testAcquireNewConnection() throws Exception {
+        EventLoopGroup group = new LocalEventLoopGroup();
+        LocalAddress addr = new LocalAddress(LOCAL_ADDR_ID);
+        Bootstrap cb = new Bootstrap();
+        ServerBootstrap sb = new ServerBootstrap();
+
+        cb.group(group)
+          .channel(LocalChannel.class);
+
+        sb.group(group)
+          .channel(LocalServerChannel.class)
+          .childHandler(new ChannelInitializer<LocalChannel>() {
+              @Override
+              public void initChannel(LocalChannel ch) throws Exception {
+                  ch.pipeline().addLast(new ChannelInboundHandlerAdapter());
+              }
+          });
+
+        // Start server
+        Channel sc = sb.bind(addr).sync().channel();
+        ChannelPoolHandler<Channel, ChannelPoolKey> handler = new TestChannelPoolHandler();
+        ChannelPool<Channel, ChannelPoolKey> pool  = new FixedChannelPool<Channel, ChannelPoolKey>(
+                cb, handler, ActiveChannelHealthChecker.instance(),
+                ChannelPoolSegmentFactories.newLifoFactory(), AcquireTimeoutAction.NewConnection, 500,
+                1, Integer.MAX_VALUE);
+
+        ChannelPoolKey key = new DefaultChannelPoolKey(addr);
+        Channel channel = pool.acquire(key).sync().getNow();
+        Channel channel2 = pool.acquire(key).sync().getNow();
+        assertNotSame(channel, channel2);
+        sc.close().sync();
+        channel.close().sync();
+        channel2.close().sync();
+        group.shutdownGracefully();
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void testAcquireBoundQueue() throws Exception {
```

Comment:
Yes good point...this is what I was thinking about...where the user can plug in their own queue :)